### PR TITLE
Incremental Static Regeneration - docs update explaining surprising behavior

### PR DESCRIPTION
**Pull request by karlkeefer**
The _requirement_ to export a `generateStaticParams` to get static cache behavior _after_ build time was _really_ surprising behavior for me, and I think others:
* https://github.com/vercel/next.js/issues/62195#issuecomment-1952091312
* https://github.com/vercel/next.js/discussions/57961#discussioncomment-8491488

Potentially this is a bug, and not something that should be fixed with documentation?
I don't understand next.js caching enough to make that determination, so instead I'm proposing these changes to docs which might be encountered by folks who are surprised by this cache behavior.

One point in favor of this being a bug: The CLI reports that a route is `SSG` enabled in the build output, but doesn't actually cache post-build page renders if this export is missing.

@awinogrodzki made a demo repo showing this behavior, as described [here](https://github.com/vercel/next.js/discussions/57961#discussioncomment-7468144).

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: c0505afb7b3c39ecbe35cc3aaf278285e6fbf9b2

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Review by samcx — APPROVED** [review]
Thank you for submitting a PR!

<picture data-single-emoji=":lgtm:" title=":lgtm:">:lgtm:</picture>.

> One point in favor of this being a bug: The CLI reports that a route is SSG enabled in the build output, but doesn't actually cache post-build page renders if this export is missing.

Good callout! We need to I think we need to improve our CLI output here.

> The requirement to export a generateStaticParams to get static cache behavior after build time was really surprising behavior for me, and I think others:

Hmm, maybe somewhat! The behavior is similar to Pages Router—if you only had a dynamic page, it would not opt you into ISR—you would have to use `getStaticPath`s. Similarly, you have to use `generateStaticParams` here.

A possible discussion to have is whether `revalidate` should just enable it, without an empty `generateStaticParams`. <picture data-single-emoji=":frog-eyes:" title=":frog-eyes:">:frog-eyes:</picture>

**Review by leerob — DISMISSED** [review]
Would love to get @delbaoliveira's opinion here before merging 🙏

**Comment by karlkeefer** [comment]
@samcx I think my confusion here may stem from he fact that I didn't use pages router before.

As more folks use next-js after app router is the recommended default, the level of surprise among users may rise ;)

Thanks for taking a look at the PR 🙌

**Comment by samcx** [comment]
> As more folks use next-js after app router is the recommended default, the level of surprise among users may rise ;)

@karlkeefer Makes sense and noted! We are working to improve the App Router's caching story which will most likely improving the Route Segment Config, so we could end up improving this to cause less confusion.

I'll defer to Delba for further review on this <picture data-single-emoji=":pr:" title=":pr:">:pr:</picture>  <picture data-single-emoji=":blob_bowing:" title=":blob_bowing:">:blob_bowing:</picture>

File: docs/02-app/02-api-reference/02-file-conventions/route-segment-config.mdx
Comment (delbaoliveira) [inline docs/02-app/02-api-reference/02-file-conventions/route-segment-config.mdx:87]:
```suggestion
> - To statically render all paths the first time they're visited, you'll need to return an empty array in `generateStaticParams`. 
```

File: docs/02-app/02-api-reference/04-functions/generate-static-params.mdx
Comment (delbaoliveira) [inline docs/02-app/02-api-reference/04-functions/generate-static-params.mdx:29]:
```suggestion
> - You must include `generateStaticParams` with an [empty array](#all-paths-at-runtime) if you want to statically render all paths at runtime. 
```

**Review by delbaoliveira — CHANGES_REQUESTED** [review]

**Comment by delbaoliveira** [comment]
Thank you for raising this @karlkeefer, I've pushed a commit to rephrase a few things. 

@samcx maybe not supplying `generateStaticParams` should be the equivalent of supplying an empty array. But I'll defer that to your team :)

**Review by delbaoliveira — APPROVED** [review]

**Review by samcx — APPROVED** [review]

**Comment by rijk** [comment]
> A possible discussion to have is whether revalidate should just enable it, without an empty generateStaticParams. :frog-eyes:

Or setting `export const dynamic = 'force-static'`. Having those options available but not actually making a route static, unless you add a `generateStaticParams` function that returns an empty array is the kind of DX that makes someone lose their mind.

**Comment by samcx** [comment]
@rijk Thanks for the feedback!

Actually not 100% with `export const dynamic = 'force-static`—taking a look <picture data-single-emoji=":frog-eyes:" title=":frog-eyes:">:frog-eyes:</picture>

**Comment by samcx** [comment]
@rijk It turns out you can use `force-static`! I'll be updating our <picture data-single-emoji=":nextjs-spin:" title=":nextjs-spin:">:nextjs-spin:</picture> <picture data-single-emoji=":docs:" title=":docs:">:docs:</picture> again <picture data-single-emoji=":blob_bowing:" title=":blob_bowing:">:blob_bowing:</picture>

**Pull request merged**
